File: frontend/src/types/station.ts
export interface CreateStation {
  id: string
  nimi: string
  namn: string
  name: string
  osoite: string
  adress: string
  kaupunki: string
  stad: string
  operaattor: string
  kapasiteet: number
  x: number
  y: number
}

export interface NewStation extends CreateStation {
  fid: number
}

export interface Station {
  fid: number
  id: string
  nimi: string
  osoite: string
  kaupunki: string
  kapasiteet: string
  x: number
  y: number
}

export interface StationDetails {
  name: string
  address: string
  numOfStartingJourney: number
  numOfEndingJourney: number
  x: number
  y: number
}

export interface GetAllStations {
  result: Station[]
  totalItems: number
}

export interface StationState {
  stations: Station[]
  totalStations: number
  stationDetails: StationDetails | null
  isLoading: boolean
  isError: any
  createdStation: NewStation | null
  isSuccess: boolean
}

File: backend/Models/Station.cs
namespace Backend.Models;

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Station
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]

    public int FID { get; set; }
    public string ID { get; set; } = null!;
    public string Nimi { get; set; } = null!;
    public string Namn { get; set; } = null!;
    public string Name { get; set; } = null!;
    public string Osoite { get; set; } = null!;
    public string Adress { get; set; } = null!;
    public string Kaupunki { get; set; } = null!;
    public string Stad { get; set; } = null!;
    public string Operaattor { get; set; } = null!;
    public int Kapasiteet { get; set; }
    public decimal X { get; set; }
    public decimal Y { get; set; }
}
